### ISSUE TITLE
conda env export lists conda packges with "_" (underscore) twice, as conda and as pip

### ISSUE THREAD

**Issue by vroetman** [conda/conda#4445]
This is seen with conda 4.3.8
`conda env export` will list packages with an underscore in their name twice, once as a conda dependency, and once as a pip dependency with the underscore changed to a dash.
For example, the "jupyter_client" package is also listed as "jupyter-client" under the pip section.

Reproduce with:
`conda create -y -n doublelist python=3.5`
`source activate doublelist`
`conda install -y jupyter_client`
`conda env export`

Environment:
```Current conda install:

               platform : linux-64
          conda version : 4.3.8
       conda is private : False
      conda-env version : 4.3.8
    conda-build version : not installed
         python version : 3.5.2.final.0
       requests version : 2.12.4
       root environment : /opt/conda  (writable)
    default environment : /opt/conda
       envs directories : /opt/conda/envs
          package cache : /opt/conda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : None
           offline mode : False
             user-agent : conda/4.3.8 requests/2.12.4 CPython/3.5.2 Linux/3.10.0-514.2.2.el7.x86_64 CentOS Linux/7.3.1611 glibc/2.17
                UID:GID : 0:0```

**Comment by kalefranz** [comment]
Is this still an issue in later versions of conda 4.3?  Latest is 4.3.18.

**Comment by cfobel** [comment]
Yes, I just tested with the following setup:

```
               platform : win-32
          conda version : 4.3.18
       conda is private : False
      conda-env version : 4.3.18
    conda-build version : 2.1.10
         python version : 2.7.12.final.0
       requests version : 2.13.0
       root environment : C:\Users\chris\Miniconda2  (writable)
    default environment : C:\Users\chris\Miniconda2
       envs directories : C:\Users\chris\Miniconda2\envs
                          C:\Users\chris\AppData\Local\conda\conda\envs
                          C:\Users\chris\.conda\envs
          package cache : C:\Users\chris\Miniconda2\pkgs
                          C:\Users\chris\AppData\Local\conda\conda\pkgs
           channel URLs : https://conda.anaconda.org/wheeler-microfluidics/win-32
                          https://conda.anaconda.org/wheeler-microfluidics/noarch
                          https://conda.anaconda.org/conda-forge/win-32
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.continuum.io/pkgs/free/win-32
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-32
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-32
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-32
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : C:\Users\chris\.condarc
           offline mode : False
             user-agent : conda/4.3.18 requests/2.13.0 CPython/2.7.12 Windows/10 Windows/10.0.14393
          administrator : unknown
```

I end up with the following entry in the `dependencies:` section of the export:

```yaml
- !!python/unicode
  'jupyter_client=4.4.0=py27_0'
```

and a duplicate listing in the `pip:` section:

```yaml
  - jupyter-client==4.4.0
```

**Note** the listing in the `pip:` section contains a **hyphen** while the listing in the `dependencies:` section contains an **underscore**.

**Comment by cfobel** [comment]
The issue stems from the following section code [`conda_env/pip_util.py`][L101]:

```python
    for pip_pkg in installed(prefix, output=output):
        if pip_pkg['name'] in conda_names and 'path' not in pip_pkg:
            continue
        installed_pkgs.add(str(pip_pkg))
```

The `pip_pkg['name'] in conda_names` check does not account for `pip list` replacing underscores in package names with hyphens.  I'm not sure of the best way to solve this...  Perhaps some type of fuzzy string matching, but that could potentially lead to false positives while still not catching all cases.

Any thoughts?


[L101]: https://github.com/conda/conda/blob/9dd5f99726e87eab1f655d5c17e656735d69da72/conda_env/pip_util.py#L101

**Comment by sachinbiradar9** [comment]
I believe this was fixed in PR #6438
